test/test_backend/routes/tst_graph.py

```diff
@@ -32,70 +38,100 @@
-test_cases_list = list(reader)
-
-extra_concept_ids = []
-hide_vocabs = ['RxNorm Extension']
-
-results_from = {
-    'date': '2023-12-18',
-    'commit': '660394d',
-}
-results = {
-    'get_connected_subgraph': {
-        'single-small': {
-            'results': [[[442793, 4034964], [4129519, 43531010]],
-                        [4174977, 43531010, 4161671, 4327944, 4221962, 4237068, 4034962, 4034964, 4294429, 4102176,
-                         45766050, 4224419, 442793, 4016047, 35626038, 35626039, 4162239, 36674752, 43022019, 3172958,
-                         4270049, 4044391, 4044392, 4129519, 4129524, 4129525, 35625724],
-                        [4174977, 43531010, 4161671, 4327944, 4221962, 4237068, 4034962, 4034964, 4294429, 4102176,
-                         45766050, 4224419, 442793, 4016047, 35626038, 35626039, 4162239, 36674752, 43022019, 3172958,
-                         4270049, 4044391, 4044392, 4129519, 4129524, 4129525, 35625724], [4252356, 44833365], {}]
-        }
-    }
-}
 
-# todo: at end, revert class, method, and file name from tst -> test
 # class TestGraph(unittest.TestCase):
 class TstGraph:
-
-    async def tst_get_connected_subgraph(self):
+    """Tests for graph.py"""
+    test_cases: List[Dict] = []
+
+    # noinspection PyPep8Naming
+    @classmethod
+    def setUpClass(cls):
+        """Set up"""
+        csv_file = StringIO(TEST_CASES_FAST.strip())
+        cases: List[Dict] = list(csv.DictReader(csv_file, delimiter=',', quotechar='"'))
+        for case in cases:
+            for k, v in case.items():
+                if k == 'codeset_ids':
+                    case[k] = v.split(',')
+        cls.test_cases = cases
+
+    async def _create_inputs__get_connected_subgraph(self):
         """Test indented_concept_list()"""
-        # todo; enable test cases from CSV
-        for case in test_cases_list:
-            codeset_ids = case['codeset_ids'].split(',')
-            sg, nodes_in_graph, orphans_not_in_graph, hidden = (
-                get_connected_subgraph(REL_GRAPH, codeset_ids, extra_concept_ids, hide_vocabs))
-            # todo: assertions
-            j = json.dumps(
-                [list(sg.edges), list(nodes_in_graph), list(orphans_not_in_graph), hidden])
-            print()
-
-
+        self.setUpClass()  # temp until tests finished
+        for test_type, test_name, codeset_ids, timeout_secs in [x.values() for x in self.test_cases]:
+            # TODO: temp
+            if test_name not in ['antibiotics 1']:
+                continue
+            input_path = THIS_INPUT_DIR / CONNECTED_SUBGRAPH_FILENAME_PATTERN.format(test_name)
+            sg, nodes_in_graph, preferred_concept_ids, orphans_not_in_graph, hidden = \
+                get_connected_subgraph(REL_GRAPH, codeset_ids, EXTRA_CONCEPT_IDS, HIDE_VOCABS)
+            #
+            normalized_hidden = {k: sorted(list(v)) for k, v in hidden.items()}
+            results = {
```

Comment:
## Nature of saved inputs
I was considering pickle vs JSON, and went with JSON. However, some of the files are big. Some of them have <100 lines, but one has almost 100k lines (~4.5MB). So I wonder if changing to pickle would be better.

Reply:
could also remove indent